azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/artifacthandler/ZIPArtifactHandler.java

```diff
@@ -6,25 +6,69 @@
 
 package com.microsoft.azure.maven.artifacthandler;
 
-//import com.microsoft.azure.maven.AbstractAzureMojo;
+import com.microsoft.azure.maven.AbstractAppServiceMojo;
+import com.microsoft.azure.maven.Utils;
 import com.microsoft.azure.maven.deploytarget.DeployTarget;
+import net.lingala.zip4j.core.ZipFile;
+import net.lingala.zip4j.exception.ZipException;
+import net.lingala.zip4j.model.ZipParameters;
+import org.apache.maven.model.Resource;
+import org.apache.maven.plugin.MojoExecutionException;
 
 import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
 
-public class ZIPArtifactHandler implements ArtifactHandler {
-    // todo qisun
-//    protected AbstractAzureMojo mojo;
+public class ZIPArtifactHandler<T extends AbstractAppServiceMojo> implements ArtifactHandler {
+    private static final String NO_RESOURCES = "No resources were specified in pom.xml or copied to stage directory.";
+    protected T mojo;
 
-//    public ZIPArtifactHandler(final AbstractAzureMojo mojo) {
-//        this.mojo = mojo;
-//    }
+    public ZIPArtifactHandler(final T mojo) {
+        this.mojo = mojo;
+    }
+
+    protected String getDeploymentStageDirectory() {
+        return Paths.get(mojo.getBuildDirectoryAbsolutePath(), this.mojo.getAppName()).toString();
+    }
 
     @Override
-    public void publish(DeployTarget target) {
+    public void publish(DeployTarget target) throws ZipException, MojoExecutionException, IOException {
+        // todo: function app zip deploy could not be tested until the sdk 1.14.0 release
+        prepareResources();
+        assureStageDirectoryNotEmpty();
         target.zipDeploy(getZipFile());
     }
 
-    protected File getZipFile() {
-        return null;
+    protected void prepareResources() throws IOException {
```

Comment:
duplicated codes also found in `FTPArtifactHandlerImpl.java`. Is it because in case the `FTPArtifactHandlerImpl.java` might be removed in future?

Reply:
OK